packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/AutoExpand.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BaseGroupingParams, HierarchyNode, HierarchyNodeHandlingParams } from "../../../HierarchyNode";
+import { GroupingHandlerResult, GroupingType } from "../Grouping";
+
+/** @internal */
+export function assignAutoExpand(props: GroupingHandlerResult): GroupingHandlerResult {
+  for (const node of props.grouped) {
+    if (Array.isArray(node.children)) {
+      const autoExpand = getGroupingAutoExpandOptionsFromParentNode(node, props.groupingType);
+      if (autoExpand === "always" || (node.children.length === 1 && !!autoExpand)) {
+        node.autoExpand = true;
```

Comment:
could `break` here?

Reply:
No, if `break` is put here, then this loop would stop on the first grouping node which has autoExpand set to true. But we need to set autoExpand for each grouping node, not only the first.